packages/xo-web/src/xo-app/pool/tab-patches.js

```diff
@@ -174,11 +181,14 @@ export default class TabPatches extends Component {
       master: { productBrand },
       missingPatches = [],
       pool,
+      poolHosts,
       userPreferences,
     } = this.props
 
     const needsCredentials = productBrand !== 'XCP-ng' && userPreferences.xsCredentials === undefined
 
+    const isSingleHost = poolHosts === undefined || size(poolHosts) === 1
```

Comment:
In which case `poolHosts` can be `undefined`?

Reply:
In case the objects haven't been fetched yet, but `filter` might actually give us an empty array, is that what you mean?

Reply:
Yes. `_.filter(undefined)` return an empty array.

Reply:
Then indeed, it's not necessary to check. But then I'd change the test to `size(poolHosts) < 2` to also disable the button if objects haven't been fetched.